packages/contracts/contracts/optimistic-ethereum/OVM/bridge/messaging/OVM_L1CrossDomainMessenger.sol

```diff
@@ -232,22 +282,35 @@ contract OVM_L1CrossDomainMessenger is
         address _target,
         address _sender,
         bytes memory _message,
-        uint256 _messageNonce,
+        uint256 _queueIndex,
         uint32 _gasLimit
     )
         override
         public
     {
-        bytes memory xDomainCalldata = _getXDomainCalldata(
-            _target,
-            _sender,
-            _message,
-            _messageNonce
+        // Verify that the message is in the queue:
+        Lib_OVMCodec.QueueElement memory element = iOVM_CanonicalTransactionChain(resolve("OVM_CanonicalTransactionChain")).getQueueElement(_queueIndex);
+
+        // Compute the transactionHash
+        bytes32 transactionHash = keccak256(
+            abi.encode(
+                address(this),
+                resolve("OVM_L2CrossDomainMessenger"),
```

Comment:
Likewise this operation is duplicated in `_sendXDomainMessage` on [line 440](https://github.com/ethereum-optimism/optimism/pull/667/files#diff-78a5e005e505965f63e037a7dc4bf7176c934a405826eb0719ba0b436e2e9134R440).